lib/arca/model.rb

```diff
@@ -0,0 +1,83 @@
+module Arca
+  class Model
+
+    # Arca::Model wraps an ActiveRecord model class and provides an interface
+    # to the collected and analyzed callback data for that class and the file
+    # path to the model class.
+    def initialize(klass)
+      @klass = klass
+      @name = klass.name
+      @callbacks = klass.arca_callback_data.dup
+      @file_path = callbacks.delete(:model_file_path)
+    end
+
+    # Array of ActiveRecord callback method symbols in a rough order of when
+    # they are used in the life cycle of an ActiveRecord model.
+    CALLBACKS = [
+      :after_initialize, :after_find, :after_touch, :before_validation, :after_validation,
+      :before_save, :around_save, :after_save, :before_create, :around_create,
+      :after_create, :before_update, :around_update, :after_update,
+      :before_destroy, :around_destroy, :after_destroy, :after_commit, :after_rollback
+    ]
+
+    # Public: ActiveRecord model class.
+    attr_reader :klass
+
+    # Public: String model name.
+    attr_reader :name
+
+    # Public: String file path.
+    attr_reader :file_path
+
+    # Public: Hash of collected callback data.
+    attr_reader :callbacks
+
+    # Public: Arca::Report for this model.
+    def report
+      @report ||= Report.new(self)
+    end
+
+    # Public: Helper method for finding the file path and line number where
+    # a method is located for the ActiveRecord model.
+    #
+    # method_symbol - Symbol representation of the method name.
+    def source_location(method_symbol)
+      source_location = klass.instance_method(method_symbol).source_location
+      {
+        :file_path => source_location[0],
+        :line_number => source_location[1]
+      }
+    rescue NameError
+      {
+        :file_path => nil,
+        :line_number => nil
+      }
+    rescue TypeError
+      {
+        :file_path => nil,
+        :line_number => nil
+      }
+    end
+
+    # Public: Hash of CallbackAnalysis objects for each callback type.
+    def analyzed_callbacks
+      @analyzed_callbacks ||= CALLBACKS.inject({}) do |result, callback_symbol|
+        Array(callbacks[callback_symbol]).each do |callback_data|
+          result[callback_symbol] ||= []
+          callback_analysis = CallbackAnalysis.new({
+            :model         => self,
+            :callback_data => callback_data
+          })
+
+          result[callback_symbol] << callback_analysis unless callback_analysis.target_file_path =~ /gems\/activerecord/
```

Comment:
Minor, but this is hard to read as it is long and wraps. I'd perhaps move the unless to the line before and indent the logic.

Also, minor but seems like you could have callback_analysis.target_file_path_active_record? or something to wrap this up.